### PR TITLE
Rename the private-chef cookbook to infra-server

### PR DESCRIPTION
**Pull request by tas50**
This makes it so private-chef doesn't show up all over the logs when we do a reconfigure or an upgrade operation in chef-server-ctl.

This also fixes a bug where I missed a resource rename in one of the subscribes.

Note: Will also need https://github.com/chef/umbrella/pull/177

Signed-off-by: Tim Smith <tsmith@chef.io>

### CODE REVIEW

File: omnibus/files/private-chef-cookbooks/Berksfile [hunk 1]
```diff
@@ -1,4 +0,0 @@
-source 'https://supermarket.chef.io'
-
-cookbook 'enterprise', git: 'https://github.com/chef-cookbooks/enterprise-chef-common', tag: 'v1.1.0'
```
Comment (tas50) [inline omnibus/files/private-chef-cookbooks/Berksfile:3]: enterprise is a dep in the main cookbook and there's no reason to get it from git other than wasting time in the build process so remove it from here.

**Comment by tas50** [comment]
https://buildkite.com/chef/chef-chef-server-master-omnibus-adhoc/builds/3059#_

**Comment by tas50** [comment]
https://buildkite.com/chef/chef-chef-server-main-omnibus-adhoc/builds/3164#ccad6c33-a577-4129-8dc4-f695d53aed71

**Comment by tas50** [comment]
https://buildkite.com/chef/chef-chef-server-main-omnibus-adhoc/builds/3222#9c211c85-ad4c-471a-80d1-7ec5c025d9a6

**Comment by jashaik** [comment]
Pipeline Information - 
Adhoc - https://buildkite.com/chef/chef-chef-server-main-omnibus-adhoc/builds/3247
Umbrella -

**Comment by lbakerchef** [comment]
Umbrella failures seem to be related to chef-manage install:

```
null_resource.chef_server_test (remote-exec): BEGIN INSTALL CHEF MANAGE ADDON
--
  |  
  | null_resource.chef_server_test (remote-exec): ++ sed 's\|^.*\/\|\|g'
  | null_resource.chef_server_test (remote-exec): + curl -vo /tmp/chef-manage_3.1.66-1_amd64.deb https://packages.chef.io/files/current/chef-manage/3.1.66/ubuntu/18.04/chef-manage_3.1.66-1_amd64.deb
  | null_resource.chef_server_test (remote-exec):   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
  | null_resource.chef_server_test (remote-exec):                                  Dload  Upload   Total   Spent    Left  Speed
  | null_resource.chef_server_test (remote-exec):   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0*   Trying 151.101.54.110...

<snip>

  | null_resource.chef_server_test (remote-exec): 100 77.5M  100 77.5M    0     0   142M      0 --:--:-- --:--:-- --:--:--  142M
  | null_resource.chef_server_test (remote-exec): * Connection #0 to host packages.chef.io left intact
  | null_resource.chef_server_test (remote-exec): ++ sed 's\|^.*\/\|\|g'
  | null_resource.chef_server_test (remote-exec): + sudo chef-server-ctl install chef-manage --path /tmp/chef-manage_3.1.66-1_amd64.deb
  | null_resource.chef_server_test: Still creating... [1m10s elapsed]
  | null_resource.chef_server_test (remote-exec): Starting Chef Infra Client, version 16.13.16
  | null_resource.chef_server_test (remote-exec): Patents: https://www.chef.io/patents
  | null_resource.chef_server_test (remote-exec): resolving cookbooks for run list: ["private-chef::add_ons_wrapper"]
  |  
  | null_resource.chef_server_test (remote-exec): ================================================================================
  | null_resource.chef_server_test (remote-exec): Error Resolving Cookbooks for Run List:
  | null_resource.chef_server_test (remote-exec): ================================================================================
  |  
  | null_resource.chef_server_test (remote-exec): Missing Cookbooks:
  | null_resource.chef_server_test (remote-exec): ------------------
  | null_resource.chef_server_test (remote-exec): No such cookbook: private-chef
  |  
  | null_resource.chef_server_test (remote-exec): Expanded Run List:
  | null_resource.chef_server_test (remote-exec): ------------------
  | null_resource.chef_server_test (remote-exec): * private-chef::add_ons_wrapper
  |  
  | null_resource.chef_server_test (remote-exec): System Info:
  | null_resource.chef_server_test (remote-exec): ------------
  | null_resource.chef_server_test (remote-exec): chef_version=16.13.16
  | null_resource.chef_server_test (remote-exec): platform=ubuntu
  | null_resource.chef_server_test (remote-exec): platform_version=18.04
  | null_resource.chef_server_test (remote-exec): ruby=ruby 2.7.4p191 (2021-07-07 revision a21a3b7d23) [x86_64-linux]
  | null_resource.chef_server_test (remote-exec): program_name=/opt/opscode/embedded/bin/chef-client
  | null_resource.chef_server_test (remote-exec): executable=/opt/opscode/embedded/bin/chef-client
  |  
  |  
  | null_resource.chef_server_test (remote-exec): Running handlers:
  | null_resource.chef_server_test (remote-exec): Running handlers complete
  | null_resource.chef_server_test (remote-exec): Chef Infra Client failed. 0 resources updated in 01 seconds
  | null_resource.chef_server_test (remote-exec): [2021-09-29T08:13:18+00:00] FATAL: Stacktrace dumped to /var/opt/opscode/local-mode-cache/chef-stacktrace.out
  | null_resource.chef_server_test (remote-exec): [2021-09-29T08:13:18+00:00] FATAL: Please provide the contents of the stacktrace.out file if you file a bug report
  | null_resource.chef_server_test (remote-exec): [2021-09-29T08:13:18+00:00] FATAL: Net::HTTPServerException: 412 "Precondition Failed"

<!--EndFragment-->
</body>
</html>
```

**Comment by lbakerchef** [comment]
Fixed previous chef manage install error.  Remaining error with external open ldap.
https://buildkite.com/chef/chef-umbrella-main-chef-server/builds/736

```
null_resource.chef_server_config (remote-exec): + sudo tee -a /opt/opscode/embedded/cookbooks/private-chef/templates/default/pedant_config.rb.erb
--
  | null_resource.chef_server_config (remote-exec): + cat /tmp/pedant_config.rb
  | null_resource.chef_server_config (remote-exec): tee: /opt/opscode/embedded/cookbooks/private-chef/templates/default/pedant_config.rb.erb: No such file or directory
```
  | null_resource.chef_server_config (remote-exec): ldap_testing true
  |  
  | null_resource.chef_server_config (remote-exec): ldap(
  | null_resource.chef_server_config (remote-exec):   "display_name": "User One",
  | null_resource.chef_server_config (remote-exec):   "first_name": "User",
  | null_resource.chef_server_config (remote-exec):   "last_name": "One",
  | null_resource.chef_server_config (remote-exec):   "common_name": "User One",
  | null_resource.chef_server_config (remote-exec):   "country": "unknown",
  | null_resource.chef_server_config (remote-exec):   "city": "unknown",
  | null_resource.chef_server_config (remote-exec):   "email": "user1@chef-server.dev",
  | null_resource.chef_server_config (remote-exec):   "username": "user1",
  | null_resource.chef_server_config (remote-exec):   "external_authentication_uid": "user1",
  | null_resource.chef_server_config (remote-exec):   "recovery_authentication_enabled": false,
  | null_resource.chef_server_config (remote-exec):   "account_name": "user1",
  | null_resource.chef_server_config (remote-exec):   "account_password": "password",
  | null_resource.chef_server_config (remote-exec):   "status": "unlinked"
  | null_resource.chef_server_config (remote-exec): )
  |  
  |  
  |  
  | Error: error executing "/tmp/terraform_554624648.sh": Process exited with status 1
```

**Comment by lbakerchef** [comment]
Fixed previous open ldap issue from within umbrella.
https://buildkite.com/chef/chef-umbrella-main-chef-server/builds/748#9df1838a-a374-4858-973e-6a5eff3114eb

See: https://github.com/chef/umbrella/pull/177

**Review by PrajaktaPurohit — APPROVED** [review]
Thank you!

**Pull request merged**